### ISSUE TITLE
FALSE NEGATIVE: `AA_ASSERTION_OF_ARGUMENTS` misses an `instanceof` check of a public method parameter

### ISSUE THREAD

**Issue by lynae99** [spotbugs/spotbugs#4305]
Hi, I found a false negative in SpotBugs 4.10.3 when a public method validates its parameter with `assert` and `instanceof`.

**Affected tool**

`SpotBugs 4.10.3`

**Affected checker**

`SpotBugs AA_ASSERTION_OF_ARGUMENTS`

**Minimal reproducer**

```java
class SpotBugsSonarQubeAAInstanceofArgument {
    public static class Subject {
        public Long convert(Object value) {
            assert value instanceof Long;
            return (Long) value;
        }
    }
}
```

**Reproduction command**

```bash
spotbugs -version
javac -d classes spotbugs-sonarqube-aa-fn-instanceof-argument.java
spotbugs -textui -effort:max -low -visitors FindArgumentAssertions classes
```

**Current behavior**

SpotBugs produces no `AA_ASSERTION_OF_ARGUMENTS` diagnostic.

**Expected behavior**

SpotBugs should report line 4 because the public method parameter is validated with an assertion that is disabled by default.
